Docker-dev-build

```diff
@@ -0,0 +1,135 @@
+ARG SPACK_VERSION=develop
+ARG OCT_VERSION=12.1
+RUN echo "Building with spack version ${SPACK_VERSION}"
+
+# Any extra packages to be installed in the host
+ARG EXTRA_PACKAGES
+RUN echo "Installing EXTRA_PACKAGES ${EXTRA_PACKAGES} on container host"
+
+# general environment for docker
+ENV SPACK_ROOT=/home/user/spack \
+	  SPACK=/home/user/spack/bin/spack \
+	  FORCE_UNSAFE_CONFIGURE=1
+
+RUN apt-get -y update
+# Convenience tools, if desired for debugging etc
+# RUN apt-get -y install wget time nano vim emacs git
+
+# From https://github.com/ax3l/dockerfiles/blob/master/spack/base/Dockerfile:
+# install minimal spack dependencies
+RUN apt-get install -y --no-install-recommends \
+              autoconf \
+              build-essential \
+              ca-certificates \
+              coreutils \
+              curl \
+              environment-modules \
+	            file \
+              gfortran \
+              git \
+              openssh-server \
+              python \
+              unzip \
+              vim \
+              wget \
+           && rm -rf /var/lib/apt/lists/*
+
+# load spack environment on login
+RUN echo "source $SPACK_ROOT/share/spack/setup-env.sh" \
+           > /etc/profile.d/spack.sh
+
+RUN adduser user
+USER user
+WORKDIR /home/user
+
+# install spack
+RUN git clone https://github.com/spack/spack.git
+# default branch is develop
+RUN cd spack && git checkout $SPACK_VERSION
+
+# # show which version we use
+RUN $SPACK --version
+
+# copy our package.py into the spack tree (and also example files)
+COPY spack/package.py $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/package.py
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus
+COPY spack/test/ $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+COPY check_buildlog.py /home/user 
+
+# Install and test serial and MPI versions of ocoptus via spack
+# # serial version
+
+RUN . $SPACK_ROOT/share/spack/setup-env.sh && \
+      # create a new environment for the serial version and activate it:
+      spack env create octopus-serial && \
+      spack env activate octopus-serial && \
+      # make a dev-build direcotry for the serial version:
+      mkdir ~/dev-build-serial && cd ~/dev-build-serial && \
+
+      wget https://octopus-code.org/down.php?file=12.1/octopus-12.1.tar.gz -O octopus-12.1.tar.gz &&\
+      tar --strip-components=1 -xf octopus-12.1.tar.gz -C ./&& \
+      # install the serial version of octopus:
+      # display specs of upcoming spack installation:
+      spack spec octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
+      # run the spack installation (adding it to the environment):
+      spack dev-build octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
```

Comment:
Comment A: I think the `spack dev-build` does not install Octopus but just the dependencies. Is that correct? (See Comment B below.)

Reply:
No, `spack dev-build` would be identical to `spack install` except that instead of spack fetching the source, we manually provide the source and the build directory. In the end all the dependencies are installed, octopus is compiled with these dependencies and the resulting octopus binary is added to the spack's database.  See [here](https://spack-tutorial.readthedocs.io/en/lanl19/tutorial_developer_workflows.html#installing-from-local-source)